consensus/state.go

```diff
@@ -2747,30 +2751,11 @@ func (cs *State) syncData() {
 		select {
 		case <-cs.Quit():
 			return
-		case proposal, ok := <-proposalChan:
+		case proposalAndFrom, ok := <-proposalChan:
 			if !ok {
 				return
 			}
-			cs.rsMtx.RLock()
-			currentProposal := cs.rs.Proposal
-			h, r := cs.rs.Height, cs.rs.Round
-			completeProp := cs.isProposalComplete()
-			cs.rsMtx.RUnlock()
-			if completeProp {
-				continue
-			}
-
-			if currentProposal == nil && proposal.Height == h && proposal.Round == r {
```

Comment:
now that we rely on consensus for proposal catchup, no need to reject proposals if they're for a previous height or if we have the full block, we should send it in the queue so that the consensus peer state is updated